docker/test/test.sh

```diff
@@ -1,22 +1,65 @@
-#!/bin/bash -xe
+#!/bin/bash -ex
 
-export CONSENSUSALGO=${CONSENSUSALGO-benchmark}
+# Important note: trying to run the stress test locally? you will need to increase your max allowed sockets open / open files
+# as shown in this stack overflow URL:
+# https://stackoverflow.com/questions/7578594/how-to-increase-limits-on-sockets-on-osx-for-load-testing
+
+echo "Cleaning up all containers, if any are running"
+docker ps -a
+docker ps -aq | xargs docker rm -fv
+sleep 3
 
 rm -rf _logs
 
+[[ -z $CONSENSUSALGO ]] && echo "Consensus algo is not set! quiting.." && exit 1
+
+# Only in Lean Helix disable the initial block height test for now
+if [[ $CONSENSUSALGO == "leanhelix" ]]; then
+ export REMOTE_ENV="true"
+fi
+
 export GIT_BRANCH=$(source ./docker/tag.sh)
 export GIT_COMMIT=$(git rev-parse HEAD)
 export SRC=/go/src/github.com/orbs-network/orbs-network-go
 
 # prepare persistent blocks for docker tests
-
 rm -rf _tmp/blocks
+
+# At the moment Lean Helix doesn't deal well with an existing blocks file
+if [[ $CONSENSUSALGO == "benchmark" ]]; then
 mkdir -p _tmp/blocks/node{1..4}
 
 cp ./test/e2e/_data/blocks _tmp/blocks/node1
 cp ./test/e2e/_data/blocks _tmp/blocks/node2
 cp ./test/e2e/_data/blocks _tmp/blocks/node3
 cp ./test/e2e/_data/blocks _tmp/blocks/node4
+fi
 
 # run docker-reliant tests
-docker-compose -f ./docker/test/docker-compose.yml up --abort-on-container-exit --exit-code-from orbs-e2e
+docker-compose -f ./docker/test/docker-compose.yml up -d
+
+export API_ENDPOINT=http://localhost:8080/api/v1/ \
+      STRESS_TEST_NUMBER_OF_TRANSACTIONS=5000 \
+      STRESS_TEST_FAILURE_RATE=20 \
+      STRESS_TEST_TARGET_TPS=100 \
+      STRESS_TEST='true' \
+      ETHEREUM_ENDPOINT=http://localhost:8545/ \
+      ETHEREUM_PRIVATE_KEY=f2ce3a9eddde6e5d996f6fe7c1882960b0e8ee8d799e0ef608276b8de4dc7f19 
+      ETHEREUM_PUBLIC_KEY=037a809cc481303d337c1c83d1ba3a2222c7b1b820ac75e3c6f8dc63fa0ed79b18 \
+      EXTERNAL_TEST='true'
+
+# the ethereum keypair is generated from the mnemonic passed to ganache on startup
+
+echo "The network has started with pre-existing (ancient) 500-some blocks"
+echo "Sleeping to allow the network to start closing new blocks.."
+echo "(So that the txpool won't throw our calls to the bin)"
+sleep 10
```

Comment:
this should be replaced with a poll of the block height metric, but you can do it in a subsequent PR - open a bug or a task on your monday board

Reply:
Agreed - I was thinking of implementing something already in this PR. I already have code for this inplace which only needs some work to make it accept arbitrary IPs like in here.

I gave up on it in the effort to not include too many moving pieces in the same PR and get the tests to run in this new constellation first.

Reply:
yes yes. good.